### PR TITLE
Feature/create multi chain

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
The PR adds the ability to construct a kinematic chain from a series of kinematic chains. This is useful in the situation where you have multiple tools that get changed out throughout the process you can create a kinematic chain for each providing a chain for the robot and a chain for the tool.

### CODE REVIEW

File: tesseract/tesseract_kinematics/include/tesseract_kinematics/kdl/kdl_utils.h [hunk 1]
```diff
@@ -148,36 +148,40 @@ struct KDLChainData
...
-
   if (!tesseract_scene_graph::parseSceneGraph(scene_graph, results.kdl_tree))
   {
     CONSOLE_BRIDGE_logError("Failed to parse KDL tree from Scene Graph");
     return false;
   }
 
-  if (!results.kdl_tree.getChain(results.base_name, results.tip_name, results.robot_chain))
+  results.chains = chains;
+  results.base_name = chains.front().first;
```
Comment (mpowelson) [inline tesseract/tesseract_kinematics/include/tesseract_kinematics/kdl/kdl_utils.h]: Should the docs mention that the first link in the chain is the base and the last is the tip?

Practically, this means that the order you list them in the SRDF matters, correct?

Comment (Levi-Armstrong) [reply tesseract/tesseract_kinematics/include/tesseract_kinematics/kdl/kdl_utils.h]: Updated.

File: tesseract/tesseract_kinematics/include/tesseract_kinematics/core/forward_kinematics_factory.h [hunk 2]
```diff
@@ -80,6 +80,23 @@ class ForwardKinematicsFactory
     return nullptr;
   }
 
+  /**
+   * @brief Create Forward Kinematics Chain Object
+   * This only need to be implemented if of type chain
+   * @param scene_graph The Tesseract Scene Graph
+   * @param chains The chains that make up the kinematic chain
+   * @param name The name of the kinematic chain
+   * @return True if init() completes successfully
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract/tesseract_kinematics/include/tesseract_kinematics/core/forward_kinematics_factory.h]: Fixed.

**Comment by Levi-Armstrong** [comment]
Replaced by PR #253

**Pull request closed without merging**